source/initial_temperature/S40RTS_perturbation.cc

```diff
@@ -1,430 +1,450 @@
+      // same for all depth splines, do it once to avoid multiple evaluations.
+      // NOTE: there is apparently a factor of sqrt(2) difference
+      // between the standard orthonormalized spherical harmonics
+      // and those used for S40RTS (see PR # 966)
+      std::vector<std::vector<double> > cosine_components(max_degree+1,std::vector<double>(max_degree+1,0.0));
+      std::vector<std::vector<double> > sine_components(max_degree+1,std::vector<double>(max_degree+1,0.0));
+
+      for (unsigned int degree_l = 0; degree_l < max_degree+1; ++degree_l)
+        {
+          for (unsigned int order_m = 0; order_m < degree_l+1; ++order_m)
+            {
+              const std::pair<double,double> sph_harm_vals = Utilities::real_spherical_harmonic(degree_l, order_m, scoord[2], scoord[1]);
+              cosine_components[degree_l][order_m] = sph_harm_vals.first;
+              sine_components[degree_l][order_m] = sph_harm_vals.second;
+            }
+        }
+
+      // iterate over all degrees and orders at each depth and sum them all up.
+      std::vector<double> spline_values(num_spline_knots,0);
+      double prefact;
+      unsigned int ind = 0;
+
+      for (unsigned int depth_interp = 0; depth_interp < num_spline_knots; ++depth_interp)
+        {
+          for (unsigned int degree_l = 0; degree_l < max_degree+1; ++degree_l)
+            {
+              for (unsigned int order_m = 0; order_m < degree_l+1; ++order_m)
+                {
+                  if (degree_l == 0)
+                    prefact = (zero_out_degree_0
+                               ?
+                               0.
+                               :
+                               1.);
+                  else if (order_m != 0)
+                    prefact = 1./sqrt(2.);
+                  else prefact = 1.0;
+
+                  spline_values[depth_interp] += prefact * (a_lm[ind] * cosine_components[degree_l][order_m]
+                                                            + b_lm[ind] * sine_components[degree_l][order_m]);
+
+                  ++ind;
+                }
+            }
+        }
+
+      // We need to reorder the spline_values because the coefficients are given from
+      // the surface down to the CMB and the interpolation knots range from the CMB up to
+      // the surface.
+      std::vector<double> spline_values_inv(num_spline_knots,0);
+      for (unsigned int i=0; i<num_spline_knots; ++i)
+        spline_values_inv[i] = spline_values[num_spline_knots-1 - i];
+
+      // The boundary condition for the cubic spline interpolation is that the function is linear
+      // at the boundary (i.e. Moho and CMB). Values outside the range are linearly
+      // extrapolated.
+      aspect::Utilities::tk::spline s;
+      s.set_points(depth_values,spline_values_inv);
+
+      // Get value at specific depth
+      const double perturbation = s(scoord[0]);
+      return perturbation;

```

Comment:
To save space I would probably just return s(scoord[0]) directly and not introduce the variable perturbation (i.e. do 'return s(scoord[0]);' ). In that case maybe making the comment a bit more descriptive would also help, e.g. 'Return value of the vs perturbation' instead of just 'Get value'.